src/targets/gpu/fuse_ops.cpp

```diff
@@ -332,6 +337,48 @@ void move_standard_front(std::vector<instruction_ref>& args)
         std::swap(*it, args.front());
 }
 
+struct find_layernorm
+{
+    template <class... Ts>
+    static auto multibroadcast_op(Ts... xs)
+    {
+        return match::name("multibroadcast")(match::arg(0)(xs...));
+    }
+
+    static auto layernorm_onnx()
+    {
+        return match::arg(0)(match::name("gpu::sub")(
+            match::arg(0)(match::any().bind("x")),
+            match::arg(1)(multibroadcast_op(match::name("gpu::reduce_mean")))));
+    }
+
+    static auto layernorm_tf()
+    {
+        return match::either_arg(0, 1)(
+            match::name("gpu::mul")(match::arg(0)(match::name("gpu::add").bind("x"))),
+            match::name("gpu::sub")(match::either_arg(0, 1)(
+                multibroadcast_op(),
+                match::name("gpu::mul")(
+                    match::arg(1)(match::name("gpu::mul")(match::arg(1)(multibroadcast_op()))
+
+                                      )))));
+    }
+
+    auto matcher() const
+    {
+        return match::name("gpu::add", "gpu::div")(match::any_of(layernorm_onnx(), layernorm_tf()));
```

Comment:
This shouldn't match `add` or `div` as these can mean very different things. Is the `div` even necessary? Dont we change it to multiply in simplify_algebra?

Reply:
We only change it to multiply in simplify_algebra if the input is a constant value.